Source/qol/itemlabels.cpp

```diff
@@ -0,0 +1,162 @@
+/**
+* @file itemlabels.cpp
+*
+* Adds item labels QoL feature
+*/
+
+#include <vector>
+#include <string>
+#include <unordered_set>
+
+#include "inv.h"
+#include "gmenu.h"
+#include "cursor.h"
+#include "common.h"
+#include "control.h"
+#include "itemlabels.h"
+#include "engine/render/cel_render.hpp"
+
+namespace devilution {
+
+namespace {
+
+struct itemLabel {
+	int id, width;
+	Point pos;
+	std::string text;
+};
+
+std::vector<itemLabel> labelQueue;
+
+bool altPressed = false;
+bool isLabelHighlighted = false;
+std::array<std::optional<int>, ITEMTYPES> labelCenterOffsets;
+bool invertHighlightToggle = false;
+
+} // namespace
+
+void ToggleItemLabelHighlight()
+{
+	invertHighlightToggle = !invertHighlightToggle;
+}
+
+void AltPressed(bool pressed)
+{
+	altPressed = pressed;
+}
+
+bool IsItemLabelHighlighted()
+{
+	return isLabelHighlighted;
+}
+
+void AddItemToLabelQueue(int id, int x, int y)
+{
+	if (altPressed == invertHighlightToggle)
+		return;
+	ItemStruct *it = &items[id];
+
+	char textOnGround[64];
+	if (it->_itype == ITYPE_GOLD)
+		std::sprintf(textOnGround, "%i gold", it->_ivalue);
+	else
+		std::strcpy(textOnGround, it->_iIdentified ? it->_iIName : it->_iName);

```

Comment:
You can avoid the string copy:
```suggestion
	const char *textOnGround;
	if (it->_itype == ITYPE_GOLD) {
		std::sprintf(tempstr, "%i gold", it->_ivalue);
		textOnGround = tempstr;
	} else {
		textOnGround = it->_iIdentified ? it->_iIName : it->_iName;
	}
```